File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/activities/SettingsActivity.java
package ru.mamykin.foreignbooksreader.ui.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.AppCompatSeekBar;
import android.support.v7.widget.SwitchCompat;
import android.view.View;
import android.widget.ImageView;
import android.widget.SeekBar;
import android.widget.TextView;

import com.arellomobile.mvp.presenter.InjectPresenter;

import butterknife.BindView;
import butterknife.OnCheckedChanged;
import butterknife.OnClick;
import ru.mamykin.foreignbooksreader.R;
import ru.mamykin.foreignbooksreader.common.UiUtils;
import ru.mamykin.foreignbooksreader.presenters.SettingsPresenter;
import ru.mamykin.foreignbooksreader.views.SettingsView;

/**
 * Страница настроек
 */
public class SettingsActivity extends BaseActivity implements SeekBar.OnSeekBarChangeListener,
        SettingsView {
    public static final String DROPBOX_DIALOG_TAG = "dropbox_dialog_tag";

    @BindView(R.id.switchBrightAuto)
    protected SwitchCompat switchBrightAuto;
    @BindView(R.id.switchNightTheme)
    protected SwitchCompat switchNightTheme;
    @BindView(R.id.seekbarBright)
    protected AppCompatSeekBar seekbarBright;
    @BindView(R.id.btnTranslateColor)
    protected View btnTranslateColor;
    @BindView(R.id.ivTranslateColor)
    protected ImageView ivTranslateColor;
    @BindView(R.id.btnDropboxLogout)
    protected View btnDropboxLogout;
    @BindView(R.id.btnTextSizeMinus)
    protected View btnTextSizeMinus;
    @BindView(R.id.btnTextSizePlus)
    protected View btnTextSizePlus;
    @BindView(R.id.tvTextSize)
    protected TextView tvTextSize;
    @BindView(R.id.tvDropboxAccount)
    protected TextView tvDropboxAccount;

    @InjectPresenter
    SettingsPresenter presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initToolbar(getString(R.string.settings), true);
        seekbarBright.setOnSeekBarChangeListener(this);
    }

    @Override
    public int getLayout() {
        return R.layout.activity_settings;
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        presenter.onBrightnessProgressChanged(progress);
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void setDropboxAccount(String account) {
        tvDropboxAccount.setText(account);
    }

    @Override
    public void showDropboxLogoutDialog() {
        UiUtils.showDialog(this, R.string.dropbox_logout_title, R.string.dropbox_logout_message,
                DROPBOX_DIALOG_TAG, () -> presenter.onDropboxLogoutPositive(), null);
    }

    @Override
    public void setContentSizeText(String text) {
        tvTextSize.setText(text);
    }

    @Override
    public void setBrightnessPos(int progress) {
        seekbarBright.setProgress(progress);
    }

    @Override
    public void setAutoBrightnessChecked(boolean enabled) {
        switchBrightAuto.setChecked(enabled);
    }

    @Override
    public void setBrightnessControlEnabled(boolean enabled) {
        seekbarBright.setEnabled(enabled);
    }

    @Override
    public void setNightThemeEnabled(boolean enabled) {
        switchNightTheme.setChecked(enabled);
    }

    @Override
    public void setTitle(String title) {
        UiUtils.setTitle(this, title);
    }

    @Override
    public void setHomeAsUpEnabled(boolean enabled) {
        UiUtils.setHomeEnabled(this, enabled);
    }

    @Override
    public void restartActivity() {
        UiUtils.restartActivity(this);
    }

    @OnCheckedChanged(R.id.switchNightTheme)
    public void onNightThemeCheckedChanged(boolean isChecked) {
        presenter.onNightThemeCheckedChanged(isChecked);
    }

    @OnCheckedChanged(R.id.switchBrightAuto)
    public void onBrightnessAutoCheckedChanged(boolean isChecked) {
        presenter.onBrightnessAutoCheckedChanged(isChecked);
    }

    @OnClick(R.id.btnDropboxLogout)
    public void onDropboxLogoutClicked() {
        presenter.onDropboxLogoutClick();
    }

    public static Intent getStartIntent(Context context) {
        return new Intent(context, SettingsActivity.class);
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/settings/SettingsActivity.kt
package ru.mamykin.foreignbooksreader.ui.settings

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.AppCompatSeekBar
import android.support.v7.widget.SwitchCompat
import android.view.View
import android.widget.ImageView
import android.widget.SeekBar
import android.widget.TextView

import com.arellomobile.mvp.presenter.InjectPresenter

import butterknife.BindView
import butterknife.OnCheckedChanged
import butterknife.OnClick
import ru.mamykin.foreignbooksreader.R
import ru.mamykin.foreignbooksreader.common.UiUtils
import ru.mamykin.foreignbooksreader.presentation.settings.SettingsPresenter
import ru.mamykin.foreignbooksreader.ui.global.YesNoDialogFragment
import ru.mamykin.foreignbooksreader.presentation.settings.SettingsView
import ru.mamykin.foreignbooksreader.ui.global.BaseActivity

/**
 * Страница настроек
 */
class SettingsActivity(override val layout: Int = R.layout.activity_settings) : BaseActivity(), SeekBar.OnSeekBarChangeListener, SettingsView {

    @BindView(R.id.switchBrightAuto)
    protected var switchBrightAuto: SwitchCompat? = null
    @BindView(R.id.switchNightTheme)
    protected var switchNightTheme: SwitchCompat? = null
    @BindView(R.id.seekbarBright)
    protected var seekbarBright: AppCompatSeekBar? = null
    @BindView(R.id.btnTranslateColor)
    protected var btnTranslateColor: View? = null
    @BindView(R.id.ivTranslateColor)
    protected var ivTranslateColor: ImageView? = null
    @BindView(R.id.btnDropboxLogout)
    protected var btnDropboxLogout: View? = null
    @BindView(R.id.btnTextSizeMinus)
    protected var btnTextSizeMinus: View? = null
    @BindView(R.id.btnTextSizePlus)
    protected var btnTextSizePlus: View? = null
    @BindView(R.id.tvTextSize)
    protected var tvTextSize: TextView? = null
    @BindView(R.id.tvDropboxAccount)
    protected var tvDropboxAccount: TextView? = null

    @InjectPresenter
    internal var presenter: SettingsPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initToolbar(getString(R.string.settings), true)
        seekbarBright!!.setOnSeekBarChangeListener(this)
    }

    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
        presenter!!.onBrightnessProgressChanged(progress)
    }

    override fun onStartTrackingTouch(seekBar: SeekBar) {

    }

    override fun onStopTrackingTouch(seekBar: SeekBar) {

    }

    override fun setDropboxAccount(account: String) {
        tvDropboxAccount!!.text = account
    }

    override fun showDropboxLogoutDialog() {
        UiUtils.showDialog(this, R.string.dropbox_logout_title, R.string.dropbox_logout_message,
                DROPBOX_DIALOG_TAG, object : YesNoDialogFragment.PositiveClickListener{
            override fun onPositiveClick() {
                presenter!!.onDropboxLogoutPositive()
            }
        }, null)
    }

    override fun setContentSizeText(text: String) {
        tvTextSize!!.text = text
    }

    override fun setBrightnessPos(progress: Int) {
        seekbarBright!!.progress = progress
    }

    override fun setAutoBrightnessChecked(enabled: Boolean) {
        switchBrightAuto!!.isChecked = enabled
    }

    override fun setBrightnessControlEnabled(enabled: Boolean) {
        seekbarBright!!.isEnabled = enabled
    }

    override fun setNightThemeEnabled(enabled: Boolean) {
        switchNightTheme!!.isChecked = enabled
    }

    override fun setTitle(title: String) {
        UiUtils.setTitle(this, title)
    }

    override fun setHomeAsUpEnabled(enabled: Boolean) {
        UiUtils.setHomeEnabled(this, enabled)
    }

    override fun restartActivity() {
        UiUtils.restartActivity(this)
    }

    @OnCheckedChanged(R.id.switchNightTheme)
    fun onNightThemeCheckedChanged(isChecked: Boolean) {
        presenter!!.onNightThemeCheckedChanged(isChecked)
    }

    @OnCheckedChanged(R.id.switchBrightAuto)
    fun onBrightnessAutoCheckedChanged(isChecked: Boolean) {
        presenter!!.onBrightnessAutoCheckedChanged(isChecked)
    }

    @OnClick(R.id.btnDropboxLogout)
    fun onDropboxLogoutClicked() {
        presenter!!.onDropboxLogoutClick()
    }

    companion object {
        val DROPBOX_DIALOG_TAG = "dropbox_dialog_tag"

        fun getStartIntent(context: Context): Intent {
            return Intent(context, SettingsActivity::class.java)
        }
    }
}